### PR TITLE
Add techincal documentation

### PR DESCRIPTION
**Pull request by d4be4st**
Task: [#263](https://app.productive.io/1-infinum/tasks/1432299)

#### Aim

Have a example of technical documentation.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Deployment
+[Semaphore](https://semaphoreci.com/APP-REPO-NAME)
+
+### Builds
+Our continuous integration tool will automatically build the environment upon each push to whatever branch.
+The build installs all dependencies and runs all the specs.
+
+### Deploying
+The `staging` branch is used for the staging environment and `master` for production.
+Whenever a branch or pull request is merged to one of those environments, after the build is finished Semaphore will try to deploy it to the environment.
```
Comment (cilim) [inline docs/README.md:23]: should we add Slack notifications documentation here as well? For what do we get notified via email, what is on Slack and what isn't notifying at all?

**Review by cilim — APPROVED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Technical Documentation
+
+<!-- Main themes -->
+### [Architecture](architecture/README.md)
+
+<!-- mostly helpers and stimulus controllers -->
+## Components
+<!-- ### [Dropdown](components/dropdown.md) -->
```
Comment (nikajukic) [inline docs/README.md:8]: I didn't quite understand what would you put here?

File: docs/architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Underline
```
Comment (nikajukic) [inline docs/architecture/README.md:1]: This should be `# Project name`?

File: docs/architecture/environment_variables.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Environment Variables
```
Comment (nikajukic) [inline docs/architecture/environment_variables.md:1]: Is this basically a copy of application.yml without values?

File: docs/architecture/staging.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+* user: **DEPLOY-USER**
+* ruby version: **RUBY VERSION**
+* node version: **NODE VERSION**
+* redis:
+  * url: **redis://URL OR SOCKET**
+  * type: local <!-- AWS ElasticCache -->
+* sidekiq
+  * systemd name: **sidekiq-APP-production.service**
+* ssh access
+  * semaphore
```
Comment (nikajukic) [inline docs/architecture/staging.md:42]: We should list other usernames here if they have access to staging?

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Technical Documentation
```
Comment (uncoverd) [inline docs/README.md:1]: @d4be4st this might be better for another PR, but I would add an example of our `developing.md` as well. We never had that on any previous projects and I think we should make an effort to make it explicit. Of course its not applicable to other projects in its current form but its a template

File: docs/architecture/staging.md (hunk 5 above)
Comment (uncoverd) [reply docs/architecture/staging.md:42]: Yep, we have it like this
```
* ssh access
  * semaphore
  * bobo
  * cilim
  * jaka
  * stef
```

File: docs/architecture/staging.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+* type: **RDS**
+* database: **Postgresql**
+* name: **DB-NAME**
+* url: **DB-URL**
+* version: **DB-VERSION**
+* extensions:
+  * plpgsql
+* Maintance and Backups:
+  * Auto minor version upgrade: enabled
+  * maintinance window: wed:03:33-wed:04:03 UTC (GMT)
```
Comment (uncoverd) [inline docs/architecture/staging.md:55]:
```suggestion
  * maintenance window: wed:03:33-wed:04:03 UTC (GMT)
```

**Review by uncoverd — APPROVED** [review]

File: docs/README.md (hunk 2 above)
Comment (d4be4st) [reply docs/README.md:8]: https://raw.githubusercontent.com/infinum/rails-pmi-atlas/master/docs/form_inputs/select.md?token=AAHNYMC6SO535KLSHBPTGZTAHDQDW

Custom components (partials, helpers, stimulus controllers) which can get big and we usually just copy/paste the existing usage without actually knowing what it does :D

File: docs/architecture/environment_variables.md (hunk 4 above)
Comment (d4be4st) [reply docs/architecture/environment_variables.md:1]: I got this idea from devops, they did this on PD Vault.
It will also be helpful when we start using docker containers where we cant use vault

File: docs/README.md (hunk 1 above)
Comment (d4be4st) [reply docs/README.md:23]: Soo something like this: @cilim 

```
### Slack channels
* #project-underline - main underline channel; client is involved
* #project-underline-backend - main underline channel for backend
* #project-underline-backend-alert - Bugsnag, Semaphore, Github alerts
```

Because emails are personal, and each person can disable/enable them how they wish...

File: docs/README.md (hunk 6 above)
Comment (d4be4st) [reply docs/README.md:1]: I wanted to do that in another PR, yes :D

**Comment by nikajukic** [comment]
@d4be4st as discussed on our meeting, I suggest you put comments for dummy values that users should enter for each attribute. It will be easier to fill out this documentation.

You have a few examples already, e.g.

**Review by nikajukic — APPROVED** [review]

**Review by melcha — APPROVED** [review]

**Comment by nikajukic** [comment]
@d4be4st can you please change CORS config to JSON as we discussed on slack?
